### ISSUE TITLE
Loss not converging for Cerebras-GPT-111M with HH dataset

### ISSUE THREAD

**Issue by chansurgeplus** [eric-mitchell/direct-preference-optimization#15]
I tried training with SFT loss on the Cerebras-GPT-111M model using HH dataset. I opted for the BasicTrainer. However, the loss didn't converge even after 30k steps, it wobbles within the same region. I tried learning rates from 0.1 up to 1e-6, all resulting in the same observation with the loss wobbling in different ranges.

Here's the command I used:
`python -u train.py model=Cerebras-GPT-111M datasets=[hh] loss=sft exp_name=anthropic_sft_Cerebras-GPT-111M gradient_accumulation_steps=2 batch_size=16 eval_batch_size=16 trainer=FSDPTrainer sample_during_eval=false`

Here goes the graph for the loss:
[WanDB Log for Cerebras-GPT-111M](https://api.wandb.ai/links/team-indigo/7n49oyx8)

Note: I experimented with Pythia-2.8b as well, using the same configuration that you have provided in the readme. I tried both FSDPTrainer and BasicTrainer, just to be sure that BasicTrainer isn't causing this. However, when trying Pythia-2.8b model, I had to incorporate QLoRA since the free resources weren't sufficient. Finally I tried FSDPTrainer for the Cerebras-GPT-111M model. These all resulted in the same observation.

Graph of the Loss for Pythia-2.8b training for very few steps: [WanDB Log for Pythia-2.8B](https://api.wandb.ai/links/team-indigo/fut4y32c)

Any clues why this is happening?

**Comment by eric-mitchell** [comment]
After looking at your learning curves, this actually looks fine to me. It's normal for the loss to only go down a little bit during SFT for HH (see our Pythia HH demo run [here](https://wandb.ai/eric_anthony_mitchell/dpo-demos/runs/i4i3ddpp?workspace=user-eric_anthony_mitchell), pic attached below). If you turn on smoothing, it should be clear that the train loss is going down. I'd also expect you to see some clear drop in eval loss.

<img width="898" alt="Screenshot 2023-07-10 at 12 46 12 PM" src="https://github.com/eric-mitchell/direct-preference-optimization/assets/56408839/0eb57575-e057-4b32-9eb6-6b9dd20a3ba9">

We originally had a small bug that made it look like the loss was going down more dramatically than it really was during SFT, but this is fixed now. After doing SFT, you should definitely see a loss drop for the DPO phase (see the DPO demo run [here](https://wandb.ai/eric_anthony_mitchell/dpo-demos/runs/og8q3euz?workspace=user-eric_anthony_mitchell)), pic attached below:

<img width="902" alt="Screenshot 2023-07-10 at 12 45 53 PM" src="https://github.com/eric-mitchell/direct-preference-optimization/assets/56408839/3e6b04c6-b574-4c26-bdd2-b8fea9a20af1">

I'm closing this issue because your results actually look as expected, but please don't hesitate to re-open if you don't feel the issue is resolved.

**Comment by yunsangju** [comment]
@eric-mitchell 
Hello.
I already have an SFT model, so I only conducted DPO training.
The train loss doesn't seem to converge as shown below.
Is the normal situation correct?
<img width="1261" alt="image" src="https://github.com/eric-mitchell/direct-preference-optimization/assets/136311766/4913f9ec-655c-4d34-b803-6c0ba9db5ffd">

**Comment by eric-mitchell** [comment]
@yunsangju thanks for the question! Can you please open a new issue where we can discuss your situation? I don't think your issue is actually related to this one (as it deals with SFT, but your issue is with the DPO stage).

**Comment by chansurgeplus** [comment]
@eric-mitchell Thanks for the quick response, I actually didn't want to respond until I got the final results. However, it seems it might take a while.

However, I'm affirmative that what you claimed is right and it makes sense according to the results that I've obtained so far. My concern was primarily based on the early results that it gave, with there's no direct evidence that the loss was converging. However, the general trend has been observed and the evaluation results proves it.

I'll post once again once I do have the final results. Thanks
